cpp/include/raft/distance/detail/kernels/gram_matrix.cuh

```diff
@@ -16,13 +16,25 @@
 
 #pragma once
 
+#include <raft/core/device_csr_matrix.hpp>
+#include <raft/core/device_resources.hpp>
 #include <raft/distance/distance.cuh>
+#include <raft/distance/distance_types.hpp>
+#include <raft/sparse/detail/cusparse_wrappers.h>
+#include <raft/sparse/distance/distance.cuh>
 
 #include <raft/linalg/detail/cublas_wrappers.hpp>
 #include <raft/linalg/gemm.cuh>
 
 namespace raft::distance::kernels::detail {
 
+template <typename math_t>
```

Comment:
We should also be exposing these publicly so users like cuml aren't using the `detail` namespace.

Reply:
Can you explain how this would break cuml? cuML is currently not using the detail namespace for these because we're exposing them through `raft::distance::kernels` public APIs.  Such as [here](https://github.com/rapidsai/cuml/blob/branch-23.04/cpp/src/svm/smosolver.cuh#L84)